pkg/db/db.go

```diff
@@ -240,6 +240,31 @@ func WithTransactionT[T any](h *DBHandler, ctx context.Context, f DBFunctionT[T]
 	return result, nil
 }
 
+type DBFunctionMultipleEntriesT[T any] func(ctx context.Context, transaction *sql.Tx) ([]T, error)
+
+// WithTransactionMultipleEntriesT is the same as WithTransaction, but you can also return and array of data, not just the error.
+func WithTransactionMultipleEntriesT[T any](h *DBHandler, ctx context.Context, f DBFunctionMultipleEntriesT[T]) ([]T, error) {
```

Comment:
this may be a little nitpick-y:
I would prefer if we had only 1 function to handle transactions e.g. calling `BegintTx`.

Idea:
We could implement `WithTransactionT` like this:
```
	res, err := WithTransactionMultipleEntriesT(h, ctx, func(ctx context.Context, transaction *sql.Tx) ([]T, error) {
		fRes, err2 := f(ctx, transaction)
		if err2 != nil {
			return nil, err2
		}
		return []T{*fRes}, nil
	})
	return &res[0], err
```
(It compiles, but I haven't run it)

Reply:
Thanks, was trying to do something similar but after a good hour fighting the compiler I decided to go with this solution. After some tweaking, got a similar solution working :)